ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc

```diff
@@ -388,13 +397,65 @@ TorchCore::processIValue (torch::jit::IValue value, GstTensorMemory *output, uns
   output_tensor = output_tensor.contiguous ();
 
   /* validate output tensor once */
-  if (configured < outputTensorMeta.num_tensors && validateOutputTensor (output_tensor, idx)) {
+  if (configured < outputTensorMeta.num_tensors
+&& validateOutputTensor (output_tensor, idx)) {
     ml_loge ("Output Tensor Information is not valid");
     return -1;
   }
 
   /** @todo avoid this memcpy */
-  std::memcpy (output->data, output_tensor.data_ptr (), output_tensor.nbytes ());
+  std::memcpy (output[idx].data, output_tensor.data_ptr (), output_tensor.nbytes ());
+  return 0;
+}
+
+/**
+ * @brief	serialize and process the output from the invoke
+ * @param[in] value IValue containing the output in tensor form
+ * @param[out] output Output tensor memory
+ * @param[inout] idx index of output
+ * @return 0 if OK. non-zero if error.
+ *         -2 if output tensor validation fails.
+ *         -3 if output is of unsupported format.
+ */
+int
+TorchCore::serializeOutput (
+    const torch::jit::IValue &value, GstTensorMemory *output, unsigned int *idx)
+{
+  /** serialize the output based on its type */
+  if (value.isTensor ()) {
+    if (processIValue (value, output, *idx)) {
+      ml_loge ("Output Tensor Information is not valid");
+      return -2;
+    }
+    (*idx)++;
+  } else if (value.isTuple ()) {
+    auto output_elements = value.toTuple ()->elements ();
+    for (auto element : output_elements) {
+      if (serializeOutput (element, output, idx)) {
+        ml_loge ("Output Tensor Information is not valid");
+        return -2;
+      }
+    }
+#ifdef PYTORCH_VER_ATLEAST_1_2_0
+  } else if (value.isList ()) {
+    c10::ArrayRef<torch::jit::IValue> output_ref_list = value.toListRef ();
+    std::vector<torch::jit::IValue> output_list (
+        output_ref_list.begin (), output_ref_list.end ());
+#else
+  } else if (value.isGenericList ()) {
+    c10::ArrayRef<torch::jit::IValue> output_list = value.toGenericListRef ();
+#endif
+    for (auto &element : output_list) {
+      if (serializeOutput (element, output, idx)) {
+        ml_loge ("Output Tensor Information is not valid");
```

Comment:
Need a better error message as well. (e.g., which parameter is invalid & why it's invalid (it shouldn't be NULL, or what has failed, ...) & how to correct it.)

Reply:
More descriptive messages have been added at the source of the error.
This location is just passing the error up the call stack and contains index information that has been added to the error message.